crates/blockchain/mempool.rs

```diff
@@ -108,10 +142,24 @@ impl Mempool {
             .transaction_pool
             .write()
             .map_err(|error| StoreError::MempoolWriteLock(error.to_string()))?;
+        self.remove_transaction_with_lock(hash, &mut tx_pool)?;
+        Ok(())
+    }
+
+    /// Remove a transaction from the pool with the transaction pool lock already taken
+    fn remove_transaction_with_lock(
+        &self,
+        hash: &H256,
+        tx_pool: &mut std::sync::RwLockWriteGuard<'_, HashMap<H256, MempoolTransaction>>,
```

Comment:
This can be simplified to:

```suggestion
        tx_pool: &mut HashMap<H256, MempoolTransaction>,
```